src/tests/ftest/pool/management_race.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+import random
+from apricot import TestWithServers
+from thread_manager import ThreadManager
+from command_utils_base import CommandFailure
+
+class PoolManagementRace(TestWithServers):
+    """
+    Epic: Create system level tests that cover pool management race boundary tests and
+          functionality.
+    Testcase:
+          DAOS-8463: Pool management integrity with race boundary conditions
+    Test Class Description:
+          Start DAOS servers, create pools and delete the pool while query and list.
+    :avocado: recursive
+    """
+
+    def del_recreate_query_and_list_pools(
+            self, del_recreate=True, pool_num=1, thread_num=1, test_loop=2):
+        """ To delete and recreate pool with the same label, to query and list pools.
+        Args:
+            del_recreate (bool): flag to delete and recreate pool with the same label.
+            pool_num (int): pool number.
+            thread_num (int): thread number.
+            test_loop (int): number of test loop to be executed.
+        """
+
+        wait_time = self.params.get("wait_time", '/run/boundary_test/*')
+        max_query_time = self.params.get("max_query_time", '/run/boundary_test/*')
+        pool_label_prefix = self.params.get("pool_label_prefix", '/run/boundary_test/*')
+        label = pool_label_prefix + str(pool_num)
+        for test_num in range(test_loop):
+            if del_recreate:
+                destroy_result = self.get_dmg_command().pool_destroy(
+                    pool=self.pool[pool_num].label.value, force=1)
```

Comment:
```suggestion
                self.pool[pool_num].destroy()
```
Why do you use dmg here? In general, it's better to use the `TestPool` instance because it cleans up the object states.

Reply:
To delete the pool with correct label to generate race condition.

Reply:
`self.pool[pool_num].destroy()` will use the correct label